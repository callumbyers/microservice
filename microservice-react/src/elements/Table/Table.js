import React from 'react'
import { 
    Table, 
    TableBody, 
    TableContainer,
    TableHead,
    TableRow,
    TableCell
} from '@material-ui/core'

const downloadStyle = {
    cursor: 'pointer'
}

const SimpleTable = ({ data, onClick, ...props }) => {
    return (
        <TableContainer>
          <Table aria-label="simple table">
            <TableHead>
              <TableRow>
                <TableCell>Name</TableCell>
                <TableCell align="right">Download</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {data.map((d) => (
                <TableRow key={d.name}>
                  <TableCell component="th" scope="row">
                    {d.name}
                  </TableCell>
                  <TableCell style={downloadStyle} onClick={() => onClick(d.url)} align="right"><h4>Download</h4></TableCell>
                </TableRow>
              ))}
            </TableBody>
          </Table>
        </TableContainer>
    )} 

export default SimpleTable;