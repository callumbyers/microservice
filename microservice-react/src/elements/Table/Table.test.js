import React from "react";
import renderer from "react-test-renderer";
import Table from "./Table";

test("renders correctly", () => {
  const data = [];
  const table = renderer.create(<Table data={data} />).toJSON();
  expect(table).toMatchSnapshot();
});
