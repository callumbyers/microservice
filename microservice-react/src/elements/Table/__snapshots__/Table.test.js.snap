// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiTableContainer-root"
>
  <table
    aria-label="simple table"
    className="MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Download
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    />
  </table>
</div>
`;
